inc/wpseo-functions.php

```diff
@@ -259,3 +259,50 @@ function wpseo_get_capabilities() {
 	}
 	return WPSEO_Capability_Manager_Factory::get()->get_capabilities();
 }
+
+
+/**
+ * Callback function to pass to the oEmbed's response data that will enable
+ * support for using the image and title set by the WordPress SEO plugin's fields. This
+ * address the concern where some social channels/subscribed use oEmebed data over OpenGraph data
+ * if both are present.
+ * 
+ * @param {array}  $data   - The oEmbed data
+ * @param {object} $post   - The current Post object
+ * @param {number} $width  - The current post's width (more applicable for attachments)
+ * @param {number} $height - The current post's height (more applicable for attachments)
+ *
+ * @see https://developer.wordpress.org/reference/hooks/oembed_response_data/ for hook info
+ *
+ * @return {array} an oEmbed array of data with modified values where appropriate.
+ */
+function wpseo_hook_oembed( $data, $post, $width, $height){
+  // data to be returned
+  $filter_data = $data;
+
+  // Look for the Yoast meta values (image and title)...
+  $yoast_title = get_post_meta( $post->ID, WPSEO_Meta::$meta_prefix . 'opengraph-title', true);
+  $yoast_image = get_post_meta( $post->ID, WPSEO_Meta::$meta_prefix . 'opengraph-image', true);
+  
+  // If yoast has a title set, update oEmbed with Yoast's title.
+  if( !empty($yoast_title) ) $filter_data['title'] = $yoast_title;
+
+  // If the a Yoast Image was set, update the oEmbed data with the Yoast Image's info
+  if( !empty($yoast_image) ){
```

Comment:
Please add some spaces to follow our codestyle guidelines.
```suggestion
  if ( ! empty( $yoast_image ) ) {
```

Reply:
I also would reverse this `if` to reduce the complexity:

```
   if ( empty( $yoast_image ) {   
      return $filter_data;
   }
```

Reply:
Spacing adjusted! For the return, I have it at the end since the function has to return a value not matter what, so I figure the double return seems repetitive? If you prefer the repetition I am happy to make the change, just let me know.